src/backend/distributed/commands/table.c

```diff
@@ -3490,6 +3490,23 @@ ErrorIfUnsupportedAlterTableStmt(AlterTableStmt *alterTableStatement)
 			{
 				if (IsForeignTable(relationId))
 				{
+					List *optionList = (List *) command->def;
```

Comment:
maybe move the logic into a function like: `ErrorIfAlterTableChangeTableNameForPostgresFDW` or such (maybe something shorter )